qds_sdk/template.py

```diff
@@ -98,31 +123,109 @@ def formatData(inputs):
     return res
                     
 class Template(Resource):
+    """
+    qds_sdk.Template is the base Qubole Template class.
+    
+    it uses /command_templates endpoint
+    """
     
     rest_entity_path = "command_templates"
     
     @staticmethod
     def createTemplate(data):
+        """
+        Create a new template.
+
+        Args:
+            `data`: json data required for creating a template
+        Returns:
+            Dictionary containing the details of the template with its ID.
+        """
         conn = Qubole.agent()
-        return conn.post(Template.rest_entity_path,data)
+        return conn.post(Template.rest_entity_path, data)
     
     @staticmethod
     def editTemplate(id, data):
+        """
+        Edit an existing template.
+
+        Args:
+            `id`:   ID of the template to edit
+            `data`: json data to be updated
+        Returns:
+            Dictionary containing the updated details of the template.
+        """
         conn = Qubole.agent()
         return conn.put(Template.element_path(id), data)
     
     @staticmethod
     def viewTemplate(id):
+        """
+        View an existing Template details.
+
+        Args:
+            `id`: ID of the template to fetch
+        
+        Returns:
+            Dictionary containing the details of the template.
+        """
         conn = Qubole.agent()
         return conn.get(Template.element_path(id))
     
     @staticmethod
-    def runTemplate(id, data):
+    def submitTemplate(id, data):
+        """
+        Submit an existing Template.
+
+        Args:
+            `id`: ID of the template to submit
+            `data`: json data containing the input_vars 
+        Returns:
+            Dictionary containing Command Object details.  
+        """
         conn = Qubole.agent()
         return conn.post(Template.element_path(id + "/run"), data)
     
     @staticmethod
+    def runTemplate(id, data):
+        """
+        Run an existing Template.
```

Comment:
Add info that it waits on the results.
Also add the info on return value, that it waits for the command to complete and prints output to stdout.